src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlSecurityToken.cs

```diff
@@ -46,7 +46,7 @@ protected void Initialize(MSIdentitySAML.SamlAssertion assertion)
 
         internal MSIdentitySAML.SamlSecurityToken WrappedSamlSecurityToken { get; }
 
-        internal MSIdentitySAML.SamlAssertion Assertion { get; private set; }
+        public MSIdentitySAML.SamlAssertion Assertion { get; private set; }
```

Comment:
I don't think this is the correct approach. We shouldn't expose a type from the Microsoft.IdentityModel namespace on a CoreWCF.IdentityModel class where it would have normally been a CoreWCF.IdentityModel type if we weren't wrapping it. My preferred approach would be to add an explicit cast operator enable casting CoreWCF.IdentityModel.Tokens.SamlSecurityToken to Microsoft.IdentityModel.Tokens.Saml.SamlSecurityToken. In the explicit cast operator, we just return the instance stored in WrappedSamlSecurityToken. Looking at the issue where this was asked for, they can then cast the CoreWCF instance to a MS.IM instance and work with those api's directly. Otherwise we're going to end up having to expose everything in some way on SamlSecurityToken, and some of the properties we would end up needing to write a whole load of new code for to support it.